### PR TITLE
Support more than one root directory

### PR DESCRIPTION
**Pull request by gizmomogwai**

### CODE REVIEW

File: source/dinotify.d [hunk 1]
```diff
@@ -277,7 +279,11 @@ public struct INotifyTree
 /// using `mask` to choose which events to watch on.
 public auto iNotifyTree(string path, uint mask)
```
Comment (DmitryOlshansky) [inline source/dinotify.d:280]: Just use type-safe variadic parameter.

string[] path..., 

See here:
https://dlang.org/spec/function.html#typesafe_variadic_functions

**Review by DmitryOlshansky — CHANGES_REQUESTED** [review]

File: source/dinotify.d [hunk 2]
```diff
@@ -273,11 +275,11 @@ public struct INotifyTree
     }   
 }
 
-/// Create a INotifyTree to recusivly establish watches in `path`,
+/// Create a INotifyTree to recusivly establish watches in `roots`,
 /// using `mask` to choose which events to watch on.
-public auto iNotifyTree(string path, uint mask)
+public auto iNotifyTree(uint mask, string[] roots...)
```
Comment (DmitryOlshansky) [inline source/dinotify.d]: Ah so it must be the last parameter?
In such case sorry for the mess, please revert to your original version and I’ll merge.

Rationale: we should not break existing code (however few users this library has).

**Review by DmitryOlshansky — CHANGES_REQUESTED** [review]

File: source/dinotify.d (hunk 2 above)
Comment (gizmomogwai) [reply source/dinotify.d]: yes ... break existing code is not so nice. i was wondering about that... how widely is dinotify used?
i am also wondering why it is, that the variadic thing needs to be last? especially if there are different types used as parameters.

Comment (DmitryOlshansky) [reply source/dinotify.d]: I’m not sure about the scope of user but it’s > 0 and that is enough for me to avoid breaking API without very strong reason.

**Review by DmitryOlshansky — APPROVED** [review]

**Comment by gizmomogwai** [comment]
its a shame that the code coverage goes down, by some percentage points, i could add a test for that, but i think that makes the test more convoluted ....

**Comment by DmitryOlshansky** [comment]
Need to publish a new version, I guess. Will do

**Comment by gizmomogwai** [comment]
👍 thanks a lot!!!

**Comment by DmitryOlshansky** [comment]
Please bump a minor version in dub and meson files

**Comment by gizmomogwai** [comment]
i think the dub registry mainly takes the version from the tag.
meson has a 0.3.0, but dub.[selections].json not. not sure if i can do a pull request for a tag?

**Comment by gizmomogwai** [comment]
also i tried to actually run the meson build, but i was not lucky. after generating the ninja files, ninja was not able to find dmd, although my path includes it.

**Comment by DmitryOlshansky** [comment]
Yeah, I totally forgot about dub being git tag driven. Cool, less patching. 
I’ll take it from here.

**Comment by gizmomogwai** [comment]
thanks a lot!

**Comment by DmitryOlshansky** [comment]
hm just tried meson + ninja works fine on Ubuntu with ldc from apt

**Comment by gizmomogwai** [comment]
ok great ... i do not have dmd in /usr/bin or something like that, but activate it on demand with the activate.fish scripts that are part of the dmd releases.

**Comment by russel** [comment]
Meson (from Debian package) build using ldc (from Debian packaging) seems to work for me.

**Comment by gizmomogwai** [comment]
may i ask how you do the build?
i guess meson builddir && cd builddir
ninja?

**Comment by russel** [comment]
I separate all my Meson build directories to ~/BuildArea. So
```
|> cd ~/BuildArea/

|> meson --libdir=lib ~/Repositories/Git/Forks/DInotify DInotify
The Meson build system
Version: 0.53.2
Source dir: /home/users/russel/Repositories/Git/Forks/DInotify
Build dir: /home/users/russel/BuildArea/DInotify
Build type: native build
Project name: dinotify
Project version: 0.4.0
D compiler for the host machine: ldc2 (llvm 1.20.1 "LDC - the LLVM D compiler (1.20.1):")
D linker for the host machine: ldc2 ld.gold 2.34
Host machine cpu family: x86_64
Host machine cpu: x86_64
Build targets in project: 2

Found ninja-1.10.0 at /usr/bin/ninja

|> cd DInotify/

|> ninja
[4/4] Linking target dinotify-test.

|> ll
total 148
drwx------  7 russel russel  4096 Mar 21 16:36  ./
drwx------ 18 russel russel  4096 Mar 21 16:36  ../
-rw-------  1 russel russel  4046 Mar 21 16:36  build.ninja
-rw-------  1 russel russel   922 Mar 21 16:36  compile_commands.json
drwx------  2 russel russel  4096 Mar 21 16:36 'dinotify@sha'/
-rwx------  1 russel russel 45616 Mar 21 16:36  dinotify-test*
drwx------  2 russel russel  4096 Mar 21 16:36 'dinotify-test@exe'/
-rw-------  1 russel russel  2992 Mar 21 16:36  dinotify-test.o
lrwxrwxrwx  1 russel russel    16 Mar 21 16:36  libdinotify.so -> libdinotify.so.0*
lrwxrwxrwx  1 russel russel    20 Mar 21 16:36  libdinotify.so.0 -> libdinotify.so.0.4.0*
-rwx------  1 russel russel 49280 Mar 21 16:36  libdinotify.so.0.4.0*
drwx------  2 russel russel  4096 Mar 21 16:36  meson-info/
drwx------  2 russel russel  4096 Mar 21 16:36  meson-logs/
drwx------  2 russel russel  4096 Mar 21 16:36  meson-private/
-rw-------  1 russel russel   136 Mar 21 16:36  .ninja_deps
-rw-------  1 russel russel   298 Mar 21 16:36  .ninja_log

|> 
```

**Comment by gizmomogwai** [comment]
thanks for those details. .. i get with dmd 2.091.0

```
> ninja
[1/1] Linking target dinotify-test.
FAILED: dinotify-test 
dmd  -of=dinotify-test 'dinotify-test@exe/source_dinotify.d.o' -L=-L--allow-shlib-undefined --main
```
seems like dmd does not understand --main but only -main .. if i run manually with -main i get more or less the same output as you.
is this a meson problem?

**Pull request merged**